### PR TITLE
Add home directory to context and fixed unit tests to support sub users

### PR DESCRIPTION
**Pull request by dekobon**
@phillipross please review if you have a free moment.

### CODE REVIEW

**Comment by phillipross** [comment]
I took a look at the code, and am I correct in my interpretation that you're using some heuristic that defines a user to be considered the owner of the account if the user begins with a "/" but not necessarily the account holder otherwise?

**Comment by dekobon** [comment]
Yes, if the username contains a "/", then it is a subuser account because slash is a reserved character.

**Comment by phillipross** [comment]
Perhaps you mean the account owner begins with a slash?  As long as it's eventually documented in the readme or something, I think any potential confusion will be minimal :grinning:

**Comment by dekobon** [comment]
Yes. That's what I meant. Yeah, I wonder where I will document this.

On Wed, Nov 18, 2015 at 6:03 PM, Phillip Ross <notifications@github.com>
wrote:

> Perhaps you mean the account owner begins with a slash? As long as it's
> eventually documented in the readme or something, I think any potential
> confusion will be minimal [image: :grinning:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joyent/java-manta/pull/69#issuecomment-157923659>.
>



-- 
-Elijah

**Comment by phillipross** [comment]
I would assume a section in the README.md would suffice for now.  There's no wiki as of yet, but I dont know if there are internal standards that joyent as that should be adhered to or what.

**Comment by dekobon** [comment]
@phillipross please have a look at the doc changes when you have a moment.

File: java-manta-client/src/main/java/com/joyent/manta/config/ConfigContext.java [hunk 1]
```diff
@@ -34,4 +34,33 @@
...
+     *         upon null mantaUser
+     */
+    static String deriveHomeDirectoryFromUser(final String mantaUser) {
+        if (mantaUser == null) {
+            return null;
+        }
+
+        final String home;
+
+        if (mantaUser.contains("/")) {
```
Comment (phillipross) [inline java-manta-client/src/main/java/com/joyent/manta/config/ConfigContext.java]: I'd recommend a slightly stronger validation here... maybe check to make sure the manta user config value contains 0 or 1 "/" characters but no more.  And also make sure the "/" is not at the beginning, or the 0-th element is going to result in an empty string (I think).

**Comment by phillipross** [comment]
After seeing the documentation, I understand how it works now... I was confused by that slash character.  I annotated the `deriveHomeDirectoryFromUser(...)` method with a comment about validation, but I'd say this is good to push as-is.  :+1:

**Pull request merged**